eventstore/views.py

```diff
@@ -166,7 +166,14 @@ def create(self, request):
 
             for statuses in request.data.get("statuses", []):
                 message_id = statuses.pop("id")
-                recipient_id = statuses.pop("recipient_id")
+
+                if "message" in statuses:
+                    recipient_id = statuses["message"].pop("recipient_id")
+                    if statuses["message"] == {}:
+                        statuses.pop("message")
```

Comment:
Not sure if this is something we'll see in practice, but what about if we have something like `{"message": {}, "recipient_id": "abc"}`, then we won't extract the recipient ID correctly here, but will have an error instead. Maybe a try/catch instead of `if "message" in statuses` might be better?

Reply:
Wouldn't the validation pass because `recipient_id` exists? It just checks if either `message.recipient_id` OR `recipient_id` exists, and in the example `{"message": {}, "recipient_id": "abc"}`, I think that's true? Either way, I don't think this is a thing we'll see in production, happy to let it be.